api/scraper/grades.js

```diff
@@ -55,13 +56,35 @@ exports.get = function (app, data, callback) {
                               'credits': parseInt(attrs.eq(4).text()),
                               'grade': attrs.eq(5).text(),
                               'exam_held': moment(attrs.eq(6).text()).format('YYYY-MM'),
-                              'result_date': ((attrs.eq(7).text() != "") ? (moment(attrs.eq(7).text()).format('YYYY-MM-DD')): null),
+                              'result_date': ((moment(attrs.eq(7).text()).isValid()) ? (moment(attrs.eq(7).text()).format('YYYY-MM-DD')): null),
                               'option': attrs.eq(8).text()};
-                if(course.course_code != "Course Code") {
-                  courses.push(course);
+
+                //Aggregating the unique semesters by checking and adding the unique EXAM_HELD values
+                var examHeld = moment(attrs.eq(6).text()).format('YYYY-MM');
+                if(!(examHeldCollection.indexOf(examHeld) > -1)) {
+                  examHeldCollection.push(examHeld);
                 }
+                courses.push(course);
               });
 
+              //Removing the column headers from the data
+              examHeldCollection.shift();
+              courses.shift();
+              var semesterWiseGrades = {};
+
+              //Consolidating the grades in a semester wise fashion
+              for(var i = 0; i < examHeldCollection.length; i++) {
+                var examHeld = examHeldCollection[i];
+                semesterWiseGrades[examHeld] = [];
+              }
+
```

Comment:
No need of this, just put in an object that has exam_held as keys and GPAs as values.